### PR TITLE
Docs: VuePress plugin to strip [V<n>] citation markers and §Sources footers at build

### PR DESCRIPTION
**Pull request by marcin-kordas-hoc**
## Summary

Adds a small markdown-it plugin wired into the existing `extendMarkdown` hook in `docs/.vuepress/config.js` that strips internal audit-harness annotations at build time so they never appear in customer-facing docs.

### Why

Internal authoring uses an audit-harness convention: factual claims are annotated inline with citation markers, and pages carry a trailing `§AuditSources` footer listing the underlying sources. These exist so the audit-harness can re-verify every claim before content ships — they are never meant for end users.

### What it does

- Removes inline citation markers — legacy `[V<n>]` **and** the current lowercase `[<prefix>_<digits>]` form (e.g. `[vrf_1]`, `[dec_3]`); bare form only, real markdown links like `[V12](url)` are preserved.
- Removes the **`§AuditSources`** footer heading and everything below it up to the next top-level (`h1`) heading. The marker is a deliberately unique token, so **legitimate `Sources` / `§ Sources` headings are preserved** (guarded by a test).
- Leaves `code_inline`, `code_block` and `fence` tokens untouched so pages that document the audit-harness syntax still render.

### Files

- `docs/.vuepress/plugins/strip-citation-markers/index.js` (plugin)
- `docs/.vuepress/plugins/strip-citation-markers/test-fixture.md` (fixture)
- `docs/.vuepress/plugins/strip-citation-markers/test.js` (Node test, 24 assertions)
- `docs/.vuepress/plugins/strip-citation-markers/test-plugin-order.js` (plugin-order integration test)
- `docs/.vuepress/config.js` (wire-up)

## Test plan

- [x] `node docs/.vuepress/plugins/strip-citation-markers/test.js` → `PASS strip-citation-markers (24 assertions)`
- [x] `node docs/.vuepress/plugins/strip-citation-markers/test-plugin-order.js` → PASS
- [x] Build confirms: zero `§AuditSources` footers and zero bare citation markers outside `<code>`/`<pre>`; `[V12](url)` rendered as a real link; inline/fenced markers preserved; legitimate `Sources` headings preserved.

---

> [!NOTE]
> **Low Risk**
> Docs-only build pipeline change with broad test coverage; main risk is accidental removal of legitimate markdown if patterns or plugin order regress.
> 
> **Overview**
> Adds a **VuePress markdown-it plugin** that strips internal audit-harness content at docs build time, and registers it in `extendMarkdown` **after** `footnotePlugin` and `includeCodeSnippet`.
> 
> The plugin mutates the parsed token stream: it removes bare inline citation markers (legacy `[V<n>]` and current `[prefix_<digits>]`, but not `[label](url)`), drops the **`§AuditSources`** section through the next `h1` or before `footnote_*` tokens, and leaves code blocks untouched. It hooks `core.ruler` before `replacements` so heading anchors see cleaned text.
> 
> **Standalone Node tests** cover fixture rendering, footnote/plugin order, and edge cases (legitimate `Sources` headings, over-stripping guards).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 084c293cda3bb019de60c1145a37ec928248fa1b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: docs/.vuepress/plugins/strip-citation-markers/index.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  // 1. Find a `Sources` heading and drop everything from it onward
+  //    (up to the next h1, if any).
+  for (let i = 0; i < tokens.length; i += 1) {
+    const t = tokens[i];
+    if (t.type === 'heading_open' && isSourcesHeading(tokens, i)) {
+      const end = findFooterEnd(tokens, i);
+      tokens.splice(i, end - i);
+      i -= 1;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply docs/.vuepress/plugins/strip-citation-markers/index.js:139]: Fixed in f6f028218

**Comment by marcin-kordas-hoc** [comment]
## Follow-up: hardened test coverage

Two extra layers of test coverage added while reviewing the plugin.

### Stronger assertions in `test.js`

Added 5 assertions that pin down edge cases the previous suite left unguarded:

| Guards against | Assertion |
|----------------|-----------|
| `\d+` weakened to `\d*` in `INLINE_CITATION_PATTERN` | literal `[V]` (no digits) must survive |
| dropping the `(?!\()` lookahead | link text `V12` stays anchored (not collapsed to `V`) |
| dropping the `i` flag on `SOURCES_HEADING_PATTERN` | lowercase `## sources` footer is also stripped |
| `t.tag === 'h1'` changed to `'h2'` in `findFooterEnd` | content under a post-Sources `# Second page` h1 survives |
| dropping `.replace(/[ \t]{2,}/g, ' ')` | `multiple markers  should collapse` must not contain a 2+ space run |

### Plugin-order coupling — `test-plugin-order.js`

`findFooterEnd` relies on `markdown-it-footnote`'s `footnote_*` tokens being present in the stream when our splice runs. Added 10 assertions across three layers:

1. **Negative control** (no footnote plugin loaded): markers + footer still stripped, but no footnote anchor appears. Anchors the "footnote tokens come from a separate plugin" assumption.
2. **Positive control** (config.js ordering: footnote first, strip last): footnotes survive AND markers/footer stripped.
3. **Rule-chain invariant**: asserts `footnote_tail` appears before `strip-citation-markers` in `md.core.ruler`. This is the actual primitive (not `md.use()` call order — both plugins hook into named ruler positions, so what matters is the resulting `core.ruler` chain). If a future `markdown-it-footnote` version moves `footnote_tail`, this assertion fires and points engineers straight at the root cause.

### Why a test, not a runtime guard

Considered an assertion inside the plugin's `md.use` registration that warns if `markdown-it-footnote` is missing. Rejected: runtime warnings in a docs-build plugin add noise and would couple the plugin to a specific footnote-plugin identity. The test artifact is strictly better — it documents the contract, fails loud on regression, and adds zero runtime overhead.

### Verification

- `node docs/.vuepress/plugins/strip-citation-markers/test.js` -> `PASS strip-citation-markers (15 assertions)`
- `node docs/.vuepress/plugins/strip-citation-markers/test-plugin-order.js` -> `PASS strip-citation-markers/test-plugin-order (10 assertions: 3 negative + 4 positive + 3 rule-chain)`
- `npx vuepress build docs` -> `success Generated static files in docs/.vuepress/dist/docs`

Commit: 4f0d2e75f

File: docs/.vuepress/plugins/strip-citation-markers/index.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ *
+ * @param {Array} tokens - Full token array.
+ * @param {number} headingOpenIdx - Index of the heading_open token.
+ * @returns {boolean} True when the heading text matches the Sources footer.
+ */
+const isSourcesHeading = (tokens, headingOpenIdx) => {
+  const inline = tokens[headingOpenIdx + 1];
+  if (!inline || inline.type !== 'inline') return false;
+  return SOURCES_HEADING_PATTERN.test(inline.content || '');
+};
```
[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply docs/.vuepress/plugins/strip-citation-markers/index.js:88]: Fixed in 410a8a2af.

File: docs/.vuepress/plugins/strip-citation-markers/test.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  .replace(/<code[\s\S]*?<\/code>/g, '')
+  .replace(/<pre[\s\S]*?<\/pre>/g, '');
+assert(
+  !/\[V\d+\]/.test(renderedWithoutCode),
+  'Expected no [V<n>] markers in rendered HTML body, but found some'
+);
+
+// 2. §Sources footer must be gone.
+assert(
+  !/Sources/i.test(renderedWithoutCode) || !/source-1/.test(rendered),
```
[the remark this answers is not in the document]
Comment (marcin-kordas-hoc) [reply docs/.vuepress/plugins/strip-citation-markers/test.js]: Tightened to `&&` — the assertion now requires both the `§Sources` heading and the footer body (`source-1`) to be removed, so it fails if either leaks. Pushed in 084c293cd.

**Comment by sequba** [comment]
I don't think we need this plugin. I'd rather rely on https://github.com/handsontable/hyperformula/pull/1678 to make sure there is no citation markers in the public documentation

**Comment by marcin-kordas-hoc** [comment]
Agreed. With #1678 now scanning `docs/api` and `docs/guide` (plus the compiled output) and failing the build on any `[V<n>]`/`[<prefix>_<digits>]` marker or `§AuditSources` footer, markers can't reach the published docs in the first place — so a render-time stripper is redundant. It's also the better guarantee: #1678 fails fast and forces the author to remove the marker at the source, whereas this plugin would silently strip it and hide that it was committed. I'll close this in favor of #1678.
